From caae695882c34978640587a8594fcd24693085c7 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 11 Sep 2013 00:03:47 +0200
Subject: [PATCH 28/41] Dynamically decide about window size based on
 information from the windowing system

Looks well on bigger displays, yet kind of sucks with network play.
---
 xlib/init.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/xlib/init.c b/xlib/init.c
index 6fad837..d9b3d91 100644
--- a/xlib/init.c
+++ b/xlib/init.c
@@ -385,6 +385,12 @@ initialize (char **argv, int argc)
       exit (0);
     }
 
+  if (MAPWIDTH == 0)
+    {
+      GAMEWIDTH = MAPWIDTH = DisplayWidth (dp, DefaultScreen (dp)) * 0.8;
+      GAMEHEIGHT = MAPHEIGHT = DisplayHeight (dp, DefaultScreen (dp)) * 0.8;
+    }
+
   if (Sync)
      XSynchronize (dp, 1);
   XSetCloseDownMode (dp, DestroyAll);
@@ -640,10 +646,8 @@ main (int argc, char **argv)
   int             c;
   XEvent          event;
   nrockets = 1;
-  GAMEWIDTH = 640;
-  GAMEHEIGHT = 460;
-  MAPWIDTH = 640;
-  MAPHEIGHT = 460;
+  GAMEWIDTH = MAPWIDTH = 0;
+  GAMEHEIGHT = MAPHEIGHT = 0;
   DIV = 1;
 #ifdef MITSHM
   shm = 1;
@@ -833,6 +837,11 @@ main (int argc, char **argv)
 #ifdef NETSUPPORT
   if (server)
     {
+      if (MAPWIDTH == 0) {
+        GAMEWIDTH = MAPWIDTH = 640;
+        GAMEHEIGHT = MAPHEIGHT = 460;
+        DIV = 1;
+      }
       init_server ();
       server_loop ();
     }
-- 
2.5.0

